ecg/ecg.py

```diff
@@ -55,28 +55,44 @@ def __write_json(self,data,fname):
         """
         with open(fname, 'w') as outfile:
             json.dump(data, outfile, indent=2)
-    
+
     #######################################################################################################
     ### RXN JSONS
     #######################################################################################################
-    
+
     def _get_biosystem_eclist(self,biosystem_json):
         """
         Converts exhaustive JGI json to only a list of ECs
         """
 
         biosystem_json = self.__load_json(biosystem_json)
-        
+
         ec_list = list()
+
+
         try:
             for ec in biosystem_json["enzymes"]:
                 try:
                     ec_list.append(ec.split("EC:")[1])
                 except:
                     pass
         except:
-            pass
-        
+            try:
+                for ec in biosystem_json["assembled"]:
```

Comment:
We should give the user the option of choosing which combination of `assembled` or `unassembled` or `both` get used for composing the `ec_list`.

Reply:
So this should probably be added as a keyword argument to this function, and any functions that it gets passed through. The question is then what we should default to in the case where it's not provided. I think that we should just default to composing the `ec_list` from the sum of the `EC:`s across all assembly types that exist in the file (i.e. exist in the `biosystem_json` dict).

Reply:
By "from the sum of the EC:s across all assembly types that exist in the file",  do you mean something like assembly_types = ["enzymes", "assembled", "unassembled"]? That is possible. Or since "assembled" and "unassembled" are keys for EC: s in metagenome, and "enzymes" is the only key used for genome, we can add metagenome or genome as arguments. Which one would be simpler and more clear?

Reply:
Yeah, that's what I mean. I think the former suggestion you have is better, because I think that there's no need to specify if you're supplying a genome or metagenome unless you want a subset of the EC numbers (like only unassembled from the metagenome for example).